File: js_sequelize/models/Task.js
const {database} = require("../database");
const {DataTypes} = require("sequelize");


module.exports.Task = database.define('Task', {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true
    },
    name: {
        type: DataTypes.STRING(100),
        allowNull: false
    },
    description: {
        type: DataTypes.TEXT,
        allowNull: false
    },
    available_from: {
        type: DataTypes.DATE,
        allowNull: false
    },
    available_to: {
        type: DataTypes.DATE,
        allowNull: true
    },
    max_points: {
        type: DataTypes.FLOAT,
        allowNull: false
    },
    course_ID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
    },
}, {
    tableName: 'tasks',
    timestamps: false
});

File: dotnet_entity_framework/Models/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace dotnet_entity_framework.Models;

[Table("tasks")]
public class Task
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime AvailableFrom { get; set; }
    public DateTime? AvailableTo { get; set; }
    public double MaxPoints { get; set; }

    [Key, ForeignKey("Course")]
    public int CourseId { get; set; }

    public Course Course { get; set; } = null!;
}
